client/src/App.js

```diff
@@ -47,168 +66,83 @@ class App extends React.Component {
-          children: this.state.children,
-          infants: this.state.infants
+          guests: this.state.guests
         })
         .then(result => {
-  
           this.handleGetBookedDates()
-          
           this.setState({
-            startDate: new Date(),
-            endDate: new Date(),
-            message: result.data
+            successMessage: result.data.validDates
           })
-  
+
           setTimeout(() =>{
             this.setState({
-              message: ''
+              successMessage: ''
             })
           }, 3000)
-  
         })
         .catch(err => {
-          console.log(err)
-        })
-     }
-
-     handleChangeStart = (startDate) => {
-
-      startDate = startDate || this.state.startDate;
-  
-      let endDate = this.state.endDate;
-  
-      if(moment(startDate).isAfter(endDate)){
-        endDate = startDate
-        this.setState({endDate})
-      }
-  
-      this.setState({startDate})
-    
-    };
-
-
-    handleChangeEnd = (endDate) => {
-      endDate = endDate || this.state.endDate;
-      const startDate = this.state.startDate
-      if(moment(startDate).isAfter(endDate)){
-        endDate = startDate
-      }
-      this.setState({endDate})
-  
-    }
+          this.setState({
+            errorMessage: err.response.data.invalidDates
+          })
 
-  
+        setTimeout(() =>{
+          this.setState({
+            errorMessage: ''
+          })
+        }, 3000)
 
-    // Guest Logic
-    // Adults
-    increaseAdultCount = () => {
-      this.setState((prevState) => ({
-        adults: prevState.adults + 1
-      }))
-    }
-    
-    decreaseAdultCount = () => {
-      if(this.state.adults > 0){
-        this.setState((prevState) => ({
-          adults: prevState.adults - 1
-        }))
-      }
-    }
-  
-    // Children
-    increaseChildCount = () => {
-      this.setState((prevState) => ({
-        children: prevState.children + 1
-      }))
-    }
-  
-    decreaseChildCount = () => {
-      if(this.state.children > 0){
-        this.setState((prevState) => ({
-          children: prevState.children - 1
-        }))
-      }
-    }
-  
-    // Infants
-    increaseInfantCount = () => {
-      this.setState((prevState) => ({
-        infants: prevState.infants + 1
-      }))
-    }
-  
-    decreaseInfantCount = () => {
-      if(this.state.infants > 0){
-        this.setState((prevState) => ({
-          infants: prevState.infants - 1
-        }))
-      }
-    }
-  
+        })
+     }
 
     // Life Cycle Methods
     componentDidMount(){
       this.handleGetBookedDates()
     }
 
+  render(){
+
+    let successModal;
     
-  
+    if(this.state.successMessage){
 
-  render(){
+      successModal = (
+        <div className="success-modal">
+          
+          <div className="success-message">{this.state.successMessage}</div>
+    
+        </div>)
+
+    } else {
+      successModal = null;
+    }
     
     return(
```

Comment:
Be careful, i think you should be fine because all your ajax stuff looks like is tucked into another component but AJAX with react is weird and @jondionowens  and I ran into an issue where we noticed it was rendering and breaking our component. We found out that is because react renders immediately before the results of the ajax request (your state) is returned. But you you remember backcastly/recastly where they had us load a "data is loading..." template page and i always wondered why the hell it was doing that. We'll its because on first render there is NO data:
https://reactjs.org/docs/faq-ajax.html

Do you see how even in the official docs they have a Loading message?
![image](https://user-images.githubusercontent.com/10136383/50799806-7c36c480-1292-11e9-98c6-030e8ff4b343.png)

Again just something to look out for in the future but i think you'll be fine